python/pip_install/parse_requirements_to_bzl/__init__.py

```diff
@@ -35,8 +35,14 @@ def parse_install_requirements(
         parser.parse(requirements_lock, constraint=False), preprocess(content)
     ):
         if parsed_line.is_requirement:
+            install_req = constructors.install_req_from_line(parsed_line.requirement)
+            if not install_req.is_pinned:
```

Comment:
What about accumulating into an `unpinned_reqs` and then raising the `RuntimeError` at the end, with the following error directive line:

`The following requirements were not pinned: {unpinned_reqs}"""))`

The current way is fail-fast on the first, which is sufficient to communicate "not fully pinned", but gives the user less info about their problem.

Reply:
👍